### PR TITLE
Fix helm pull chart version from ace.yaml

### PR DESCRIPTION
**Pull request by tomsolem**
Ensure that the version set in the ace.yaml file is used when Jenkins pulls charts from ace repo.

### CODE REVIEW

File: vars/acePushConfigToGit.groovy [hunk 1]
```diff
@@ -7,14 +7,17 @@ void call(Map opts = [:]) {
...
 
   if (!gitopsRepo) {
     error('[ace] No gitops repo specified, dying.')
   }
 
   print "[ace] gitops repo strategy is ${strategy}"
 
-  helmPullChart(target.chart, opts)
+
+  helmPullChart(helmChart, helmChartVersion, opts)
```
Comment (Starefossen) [inline vars/acePushConfigToGit.groovy:20]: If you put the version inside the `opts` the change would be backwards compatible. Not sure if there are other instances using the `helmPullChart` directly?

Comment (tomsolem) [reply vars/acePushConfigToGit.groovy]: Change to set default values so that is should now be backward compatible.

Comment (bjorges) [reply vars/acePushConfigToGit.groovy]: What happens if someone puts a helmChartVersion in opts? This is where it should go I think, not as a separate option.

File: vars/helmPullChart.groovy [hunk 2]
```diff
@@ -1,7 +1,8 @@
-void call(String chart, Map opts = [:]) {
+void call(String chart, Map opts = [:], String helmChartVersion = '') {
   Map containers = opts.containers ?: [:]
   String helmContainer = containers.helm ?: ''
   List<String> helmOpts = opts.helmOpts ?: ["--entrypoint=''"]
```
Comment (Starefossen) [inline vars/helmPullChart.groovy:4]: You could also add this on line 5 like this and remove the argument from the function signature:

```
String helmChartVersion = opts.helmChartVersion ?: ''
```

File: vars/helmPullChart.groovy [hunk 3]
```diff
@@ -1,7 +1,8 @@
...
+  
 
   println "[ace] - Got containers - ${containers}"
```
Comment (bjorges) [inline vars/helmPullChart.groovy:8]: Not able to make a comment on hte line below for some reason.. 
In the definition of aceContainerWrapper it takes an option helmOpts. This is never used.. What was/is the intended way this should work, @ekarlso ?

**Review by bjorges — CHANGES_REQUESTED** [review]
the opts->helmOpts setting is already there, I think this should be used as a general way to pass options to the helm pull command.

File: vars/helmPullChart.groovy (hunk 2 above)
Comment (tomsolem) [reply vars/helmPullChart.groovy:4]: As I understand the `opts` is comming from the jekinsfile. I.e. 

```
opts = [
  buildAgent: 'jenkins-golang',
  aceFile: 'ace.yaml'
]
```

In ace.yaml we have the chart version that the helm command need. How do we then get the chart version value from ace.yaml file to opts file? Or do we then need to set the chart version in both opts and ace.yaml file?

File: vars/helmPullChart.groovy (hunk 3 above)
Comment (tomsolem) [reply vars/helmPullChart.groovy:8]: :pray: can @bjorges or @Starefossen explain how the opts are connected to the ace.yaml file? Opts are defined in the jekinsfile in the source code repository. 

:man_shrugging: Is there any specification on how to use the opts combined with ace.yaml file? And how about the `opts.helmOpts` :eyes:. I hope we can get the chart and chart version defined in the ace.yaml file and use this to pull the chart with correct version from the chart repo.

**Comment by bjorges** [comment]
Da skal #52 være merget og denne kan lukkes

**Pull request closed without merging**